### ISSUE TITLE
Add typst support to plotters

### ISSUE THREAD

**Issue by Kreijstal** [plotters-rs/plotters#513]
### What is the feature ?
This is a pure rust alternative of LaTeX

### (Optional) Why this feature is useful and how people would use the feature ?
You can write code that contains mathematical syntax and fonts

**Comment by AaronErhardt** [comment]
I do not quite understand how this should work. Do you mean that plotters should be usable from typst code or that typst code should be usable from plotters?

**Comment by Kreijstal** [comment]
well, in matplotlib you can have a `latexified` title, here you should be able to do the same, to put equations in your plots

**Comment by nil-is-lin** [comment]
If typst could be integrated with this package, it would be a very elegant writing tool. Of course, typst itself supports math modes in latex, so naturally you can write latexed text to graphs.

**Comment by jonaspleyer** [comment]
Having read this comment
> well, in matplotlib you can have a `latexified` title, here you should be able to do the same, to put equations in your plots

my understanding is that we would like to generate code which when imported in a typst document is rendered to display the image which was drawn by `plotters`. This would mean to write a new backend which can export to the `.typ` format just like svg or the bitmap backend.

I like the idea but it is not simple at all but have some thoughts on this.

## Styling, Set and Show Rules
Typst has a very simple and powerful syntax to style even basic building blocks such as rectangles, circles, lines etc. While this is a great tool for writers to quickly change layout and theming of their document, this makes it really hard when generating code. It is possible to overwrite the "general" styling arguments for each geometric shape (circle, line, rectangle, etc.) but this would mean that when generating code which can be rendered inside typst documents, we would have to specify every (even optional) argument of these functions to obtain repeatable builds which display identical even in different documents.

## Practicability
Typst is a very fast moving language and features are being added constantly. I have only partially played around with it and have not seen many breaking changes 

## (Why) Do we want this?
### Detour to `matplotlib` and `pgf`
Let's compare this feature to the option of `matplotlib` to generate `pgf` files which can be included in LaTeX documents. It is not very easy to generate nice plots from inside of a LaTeX document without the use of other tools (such as lualatex etc.). `matplotlib` offers a vast choice of plotting options. The reason to generate `pgf` files from matplotlib is to match the style of the LaTeX document as closely as possible. This means (although often not true in practice) that for example fonts should be invariant of scaling of the overall figure and thus always match the font-style and size of the text even when we resize the overall figure.

### Differences
However, typst is much more flexible than LaTeX when it comes to scripting. This means (and people are already doing this) that we can generate (some) plots by writing pure typst code. While performance may be an issue in the future, it is clear that the flexibility of typst does not exhibit the same kind of problems which LaTeX does. Thus I find it questionable if this desired feature will really have a meaningful impact and be used at all.

## TLDR
- I see some technical problems (and choices to be made in this regard).
- I do not know if this feature would be used at all since typst scripting is already very powerful

**Comment by st3r4g** [comment]
> This would mean to write a new backend which can export to the `.typ` format just like svg or the bitmap backend.

No, I think it would work in the following way:

1. ask typst to render a single math expression into svg
2. include the svg element into plotters graph
3. generate the final picture with the already existing plotters backends

Since typst is also written in Rust, Step 1 could be implemented by Rust calls into the typst library. However, one has to see if the required functionality is exposed and if the interfaces are stable enough.
An easier way is to use the CLI interface (as matplotlib does for latex). Plotters would prepare the following minimal document for a single expression (standalone.typ):
```
#set page(width: auto, height: auto, margin: 0pt)
$ E = m c^2 $
```
Then invoke the typst binary to compile it to svg:
```
typst compile -f svg standalone.typ -
```
And finally use the output as an "external" element that can be placed in a graph at a given location.

I think this is actually quite feasible.

**Comment by jonaspleyer** [comment]
Looking back at this issue again I now understand the original question. My answer was probably misguided by something else. @st3r4g I agree with your approach.
